user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -119,7 +121,7 @@ protected void createOrUpdate(Reconciliation reconciliation, KafkaUser kafkaUser
         log.debug("{}: Updating User", reconciliation, userName, namespace);
         Secret desired = user.generateSecret();
         CompositeFuture.join(
-                scramShaCredentialOperator.reconcile(user.getName(), desired != null ? desired.getData().get("password") : null),
+                scramShaCredentialOperator.reconcile(user.getName(), (desired != null && desired.getData().get("password") != null) ? new String(Base64.getDecoder().decode(desired.getData().get("password")), Charset.forName("US-ASCII")) : null),
```

Comment:
I think this is long enough to be worth pulling into its own statement, or at least spread over multiple lines.